src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/LatexQuoteInspection.kt

```diff
@@ -0,0 +1,27 @@
+package nl.hannahsten.texifyidea.inspections.latex.probablebugs
+
+import com.intellij.codeInspection.ProblemHighlightType
+import nl.hannahsten.texifyidea.inspections.TexifyRegexInspection
+import java.util.regex.Pattern
+
+/**
+ * @author Michael Milton
+ */
+class LatexQuoteInspection : TexifyRegexInspection(
+        inspectionDisplayName = "Incorrect quotation",
+        inspectionId = "AsciiQuotes",
+        errorMessage = { """"ASCII quotes" were used instead of ``LaTeX quotes''""" },
+        highlight = ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+        pattern = Pattern.compile("""".+?"|'.+?'"""),
```

Comment:
Another false positive that was also surprising to me, if you look at the map here: https://europeisnotdead.com/european-quotation-marks/amp/
it says that in Swedish and Finnish the use of closing quotes on both sides is actually _correct_!

```latex
\documentclass[swedish]{article}
\usepackage[autostyle]{csquotes}
\begin{document}
To this end, we "bundled" the emphasis blockchain. % False positive
\end{document}

```

German is also fun, and considering that most LaTeX users are German, we should probably support the Babel way of German quotes:

```latex
\documentclass{article}
\usepackage[german]{babel}
\begin{document}
This is "`early"' calculated. % False positive
\end{document}
```

Reply:
Okay I can see this will be a nightmare unless I **always** correct to the `csquotes` shortcut ``"`something"'`` as demonstrated here: https://tex.stackexchange.com/a/142292/169711. I think `babel` isn't actually required to do this, only `csquotes`, but it won't get properly localised without `babel`. Can you explain how I can get it to add `csquotes` automatically when I correct the punctuation?

Reply:
Oh yuck, no it's quite complicated. You need to load babel, and csquotes, and run the quote defining commands before you get a truly portable set of quotes. I'm not really sure what to do here considering there's no easy way to solve this generally.

Reply:
Yeah, there's no nice generic solution because it's not really possible in all cases to tell what people want.

What I would do, is simply limit the inspection to all the cases where we are relatively sure it is a true positive.
That way, not much people will get irritating false positives, but still we catch like almost all cases (because who writes LaTeX in Finnish?).

So, I would extend the `cancelIf` argument to check for presence of swedish/finnish, and change the regex to something like 
```
"[^`"']+?"
```
and also disable when `\MakeOuterQuote` is used (or perhaps when csquotes is used at all, not sure), etc. 
You can look in other inspections how that can be checked, e.g. LatexVerticallyCenteredColonInspection

Reply:
Actually after further research, I'm fairly sure that ``` ``foo'' ``` is the only correct base LaTeX quote style, and the other styles are all babel macros. So I think I could fairly easily keep the same check I have (albeit fixed), but just cancel the inspection if babel is loaded.

Reply:
Why? Babel is like almost always loaded so that would really exclude a lot. I don't see why using `"text"` with babel would suddenly be correct if I am using English?

Reply:
Sorry, I meant that we disable it if `\MakeOuterQuote` has been used.

Reply:
Ah, yes, completely agree.